src/services/rename.ts

```diff
@@ -43,7 +48,13 @@ namespace ts.Rename {
         }
 
         if (isStringLiteralLike(node) && tryGetImportFromModuleSpecifier(node)) {
-            return options && options.allowRenameOfImportPath ? getRenameInfoForModule(node, sourceFile, symbol) : undefined;
+            return preferences.allowRenameOfImportPath ? getRenameInfoForModule(node, sourceFile, symbol) : undefined;
+        }
+
+        // Disallow rename for elements that would rename across `*/node_modules/*` packages.
+        if (wouldRenameAcrossExternalLibrary(sourceFile, symbol, typeChecker, preferences)) {

```

Comment:
As in the message, unless we use "external library" with the same meaning elsewhere, I'd say something like `wouldRenameInOtherNodeModules`.